clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -375,7 +376,7 @@ public void testMetadataFetchOnStaleMetadata() throws Exception {
         try {
             producer.send(extendedRecord, null);
             fail("Expected KafkaException to be raised");
-        } catch (KafkaException e) {
+        } catch (PartitionOutOfRangeException e) {
```

Comment:
Just a minor thing here: I think it'd be nicer to use the ExpectedException rule here (https://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html) instead of empty catch.

Reply:
Thanks Viktor! I'll get started on a KIP